test/unit/pcv/ERC20TokemakPCVDeposit.test.ts

```diff
@@ -0,0 +1,188 @@
+
+    it('succeeds if requestWithdrawal has been called', async function () {
+      // deposit in the PCVDeposit
+      await this.token.mint(this.deposit.address, this.depositAmount);
+      await this.deposit.deposit();
+
+      const userBalanceBefore = await this.token.balanceOf(userAddress);
+
+      // request withdrawal
+      await this.deposit.connect(impersonatedSigners[governorAddress]).requestWithdrawal(this.depositAmount, {});
+
+      // withdrawing should take balance back to 0
+      expect(await this.deposit.balance()).to.be.equal(this.depositAmount);
+      await this.deposit
+        .connect(impersonatedSigners[pcvControllerAddress])
+        .withdraw(userAddress, this.depositAmount, {});
+      expect(Number((await this.deposit.balance()).toString())).to.be.equal(0);
+
+      const userBalanceAfter = await this.token.balanceOf(userAddress);
+
+      expect(userBalanceAfter.sub(userBalanceBefore).toString()).to.be.equal(this.depositAmount);
+    });
+  });
+
+  describe('Withdraw ERC20', function () {
+    it('reverts if not PCVController', async function () {
+      await expectRevert(
+        this.deposit
+          .connect(impersonatedSigners[userAddress])
+          .withdrawERC20(this.token.address, userAddress, this.depositAmount, {}),
+        'CoreRef: Caller is not a PCV controller'
+      );
+    });
+
+    it('succeeds if called as PCVController', async function () {
+      // deposit in PCVDeposit
+      await this.token.mint(this.deposit.address, this.depositAmount);
+      await this.deposit.deposit();
+
+      expect(await this.deposit.balance()).to.be.equal(this.depositAmount);
+
+      // send half of pool tokens somewhere else
+      await this.deposit
+        .connect(impersonatedSigners[pcvControllerAddress])
+        .withdrawERC20(this.pool.address, userAddress, this.depositAmount.div(toBN('2')), {});
+
+      // balance should also get cut in half
+      expect(await this.deposit.balance()).to.be.equal(this.depositAmount.div(toBN('2')));
+      // user should have received the pool tokens
+      expect(await this.pool.balanceOf(userAddress)).to.be.equal(this.depositAmount.div(toBN('2')));
+    });
+  });
+
+  describe('Claim Rewards', function () {
+    it('reverts if paused', async function () {
+      await this.deposit.connect(impersonatedSigners[governorAddress]).pause({});
+      await expectRevert(
+        this.deposit.claimRewards(
+          '1', // chainId
+          '83', // cycle
+          userAddress, // wallet,
+          '524310123843078144915', // amount
+          '27', // v
+          '0x4fa17a99b5c319727c9a7c846112902f88d8a261049e70737cd1d60e52609c50', // r
+          '0x44ab278ba36dedbd9b06ceb6c60f884dbaeacb8b3ac4043b901267a2af02ef6f', // s
+          {}
+        ),
+        'Pausable: paused'
+      );
+    });
+
+    it('should claim TOKE rewards & leave them on the deposit', async function () {
+      const rewardAddress = '0x4eff3562075c5d2d9cb608139ec2fe86907005fa';
```

Comment:
in the unit test that use the mock rewards contract, it will always work. I removed the address parameter (joey review), so I'll mark this comment as resolved